lib/full-site-editing/block-templates.php

```diff
@@ -93,18 +95,42 @@ function _gutenberg_get_template_files( $template_type ) {
 				// Subtract ending '.html'.
 				-5
 			);
-			$template_files[] = array(
-				'slug'  => $template_slug,
-				'path'  => $template_file,
-				'theme' => $theme_slug,
-				'type'  => $template_type,
+			$template_files[] = _gutenberg_conditionally_add_template_part_section_info(
+				array(
+					'slug'  => $template_slug,
+					'path'  => $template_file,
+					'theme' => $theme_slug,
+					'type'  => $template_type,
+				)
 			);
 		}
 	}
 
 	return $template_files;
 }
 
+/**
+ * Attempts to add the template part's section information from the
+ * experimental-theme.json file to the input array.
+ *
+ * @param array $template_info Template to add information to (requires 'type' and 'slug' fields).
+ *
+ * @return array Template.
+ */
+function _gutenberg_conditionally_add_template_part_section_info( $template_info ) {
+	if ( 'wp_template_part' === $template_info['type'] && is_readable( locate_template( 'experimental-theme.json' ) ) ) {
```

Comment:
I'm surprised we don't have any functions to check for the theme.json file... and we don't have any functions to read it 🤔 It would be great to load the file only once per request. We are checking and reading, decoding the file for each template part.

We have a function called `gutenberg_experimental_global_styles_has_theme_json_support`, which checks for the existence for the `experimental-theme.json` file. Later we should remove the `global_styles` prefix and move it to a different file so it acts as a global function for checking for theme JSON support. (It's not something we should tackle in this PR, just a thought)

Reply:
Not sure why we would be checking for theme.json here?

Reply:
> Not sure why we would be checking for theme.json here?

@mtias - In order to allow themes to define a starting value for the term for the template parts they provide.  @youknowriad suggested using the theme.json and it seems like a good way to go about it.  Alternatives i could think of  would be going by slug (filename) prefix ( `header-*` ) or maybe adding some special markup in the file that doesnt correspond to an element or block.. but it seems like allowing them to define the term in the theme.json seems best of those options.  Open to other ideas though!

> It would be great to load the file only once per request. We are checking and reading, decoding the file for each template part.

@david-szabo97 good point!  I think that would make sense for now, but if our goal in the future is to start supporting items from non-active themes that might be a bit more difficult.  Will see if I can update this to only check and decode it once.

Reply:
> We are checking and reading, decoding the file for each template part.

Only decoding once per request now. 👍